application/comit_node/src/swap_protocols/rfc003/role.rs

```diff
@@ -0,0 +1,137 @@
+use crate::swap_protocols::{
+    self,
+    asset::Asset,
+    rfc003::{ledger::Ledger, SecretHash},
+};
+use std::fmt::Debug;
+
+pub trait Role: Send + Sync + Debug + Clone + 'static {
+    type AlphaLedger: Ledger;
+    type BetaLedger: Ledger;
+    type AlphaAsset: Asset;
+    type BetaAsset: Asset;
+    type AlphaRedeemHtlcIdentity: Send
+        + Sync
+        + Clone
+        + Debug
+        + PartialEq
+        + Into<<Self::AlphaLedger as swap_protocols::Ledger>::Identity>;
+
+    type AlphaRefundHtlcIdentity: Send
+        + Sync
+        + Clone
+        + Debug
+        + PartialEq
+        + Into<<Self::AlphaLedger as swap_protocols::Ledger>::Identity>;
+
+    type BetaRedeemHtlcIdentity: Send
+        + Sync
+        + Clone
+        + Debug
+        + PartialEq
+        + Into<<Self::BetaLedger as swap_protocols::Ledger>::Identity>;
+
+    type BetaRefundHtlcIdentity: Send
+        + Sync
+        + Clone
+        + Debug
+        + PartialEq
+        + Into<<Self::BetaLedger as swap_protocols::Ledger>::Identity>;
+
+    type Secret: Send + Sync + Clone + Into<SecretHash> + Debug + PartialEq;
+}
+
+#[derive(Debug)]
+pub struct Initiation<R: Role> {
+    pub alpha_ledger_refund_identity: R::AlphaRefundHtlcIdentity,
+    pub beta_ledger_redeem_identity: R::BetaRedeemHtlcIdentity,
+    pub alpha_ledger: R::AlphaLedger,
+    pub beta_ledger: R::BetaLedger,
+    pub alpha_asset: R::AlphaAsset,
+    pub beta_asset: R::BetaAsset,
+    pub alpha_ledger_lock_duration: <R::AlphaLedger as Ledger>::LockDuration,
+    pub secret: R::Secret,
+}
+
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub enum ActionKind<Accept, Decline, Deploy, Fund, Redeem, Refund> {
+    Accept(Accept),
+    Decline(Decline),
+    Deploy(Deploy),
+    Fund(Fund),
+    Redeem(Redeem),
+    Refund(Refund),
+}
+
+impl<Accept, Decline, Deploy, Fund, Redeem, Refund>
+    ActionKind<Accept, Decline, Deploy, Fund, Redeem, Refund>
+{
+    pub fn name(&self) -> String {
+        use self::ActionKind::*;
+        match *self {
+            Accept(_) => String::from("accept"),
+            Decline(_) => String::from("decline"),
+            Deploy(_) => String::from("deploy"),
+            Fund(_) => String::from("fund"),
+            Redeem(_) => String::from("redeem"),
+            Refund(_) => String::from("refund"),
+        }
+    }
+}
+
+pub trait Actions {
+    type ActionKind;
+
+    fn actions(&self) -> Vec<Self::ActionKind>;
+}
+
+#[cfg(test)]
+pub mod test {
```

Comment:
❓Why having this test mod if there are no tests in it?

Reply:
It exports things that are useful for tests .